test/Microsoft.Extensions.Internal.Test/ObjectMethodExecutorTest.cs

```diff
@@ -427,6 +511,17 @@ public ValueTask<string> ValueTaskOfReferenceType(string result)
             public void MethodWithMultipleParameters(int valueTypeParam, string referenceTypeParam)
             {
             }
+
+            public FSharpAsync<string> FSharpAsyncMethod(string parameter)
```

Comment:
@panesofglass Thanks for the offer! As it happens I already implemented the basics of this yesterday - it's now in this pull request https://github.com/aspnet/Mvc/pull/6231/files in case you have any suggestions for making it more F#-idiomatic.